### ISSUE TITLE
Shortcodes blank WC2 to WC3 update (with fix)

### ISSUE THREAD

**Issue by netrosis** [woocommerce/woocommerce#17708]
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [x] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. Be running a Woocommerce 2.x site and using a shortcode like [featured_products columns="4"]
2. Update Woocommerce to v3.x and run the database update
3. Check your page running the shortcode. Nothing will display.

This is not an isolated issue, first results when googling: [https://wordpress.org/support/topic/featured-products-shortcode-not-working-after-update/](https://wordpress.org/support/topic/featured-products-shortcode-not-working-after-update/)

## Expected/actual behavior

When I follow those steps, I see...
Nothing

I was expecting to see...
A listing of featured products

## How to fix

* If you change something about the shortcode, it will show products again.
* If you clear out woocommerce transients, it will work again.

It seems that the transients for shortcodes are not cleared or updated during the update.
With the change for featured products and product visibility moving to taxonomies in Woocommerce 3, it seems like something there has caused something in the old transient to not work with WC3.

Quite simply, add in something during the update from WC2 to WC3 to clean out the shortcode transients. They should automatically rebuild themselves properly if called again.

**Comment by GlennMartin1** [comment]
Are you aware that featured_products is deprecated? (though it should still work, you should revise your shortcodes)

https://docs.woocommerce.com/document/woocommerce-shortcodes/

**Comment by kloon** [comment]
As @GlennMartin1 mentioned, the featured_products shortcode is no longer supported anymore. We moved it over to a new global products shortcode which can achieve the same functionality.

Just replace your shortcode with [products visibility="featured"]

I have also tested the backward compatibility on my local install by installing an old version of WooCommerce, creating a page with the featured_products shortcode and then upgrading to the lastest version and the page was still displaying fine on my end. That said it would be wise to change your shortcode rather as the backward compatibility will be removed in a future release.
